File: src/main/java/com/pan/Hello.java
package com.pan;

public class Hello {
    public static void main(String[] args) {
//        System.out.println("Hello world");
//        new Person().hello();
//        大寫的物件為參照資料型態
//        String s = new String("abcd")
        String s = "abcd"; //null 空白參照資料型態
//        NullPointerException記憶體為Null，沒有指到任何物件!!
        System.out.println(s.charAt(3)); //index
        System.out.println(s.length()); //長度
        System.out.println(s.substring(3));//取子字串 ==> 從第3index之後
        System.out.println(s.substring(1,4));//取子字串 ==> 從第1index取到4index之前
        Person p = new Person();
        p.hello();
    }
}

File: src/main/kotlin/com/pan/Hello.kt
package com.pan

fun main() {
    var s:String? = "abcde" //問號允許放null
    s = null
    println(s!!.length) //能為null則加?，確認要自行處理exception則加上!!
    println(s?.get(3)) //取到某index的值
    println(s?.substring(3)) //子字串
//    kotlin處理null pointer較厲害



//    println("Hello Kotlin")
//    Human().hello()

//    物件
    val h = Human()
    h.hello()
}

class Human{
    fun hello(){
        println("Hello Kotlin")
    }
}

//fun main(args: Array<String>) {
//
//}

